src/Compilers/CSharp/Portable/FlowAnalysis/NullableWalker.cs

```diff
@@ -4453,26 +4469,27 @@ private bool HasTopLevelNullabilityConversion(TypeWithAnnotations source, TypeWi
             Location location = null)
         {
             Debug.Assert(!IsConditionalState);
-            Debug.Assert(node != null);
-            Debug.Assert(operandOpt != null || !operandType.HasNullType);
+            Debug.Assert(convertedNode != null);
             Debug.Assert(targetTypeWithNullability.HasType);
             Debug.Assert((object)target != null || assignmentKind != AssignmentKind.Argument);
             Debug.Assert(conversion.Kind == ConversionKind.ExplicitUserDefined || conversion.Kind == ConversionKind.ImplicitUserDefined);
 
             TypeSymbol targetType = targetTypeWithNullability.Type;
-            location ??= node.Syntax.GetLocation();
+            location ??= conversionOpt?.Syntax.GetLocation() ?? convertedNode.Syntax.GetLocation();
```

Comment:
>location ??= conversionOpt?.Syntax.GetLocation() ?? convertedNode.Syntax.GetLocation(); [](start = 12, length = 87)

Please make `location` non-optional. Then it looks like this statement can be removed.